crates/wasm-pkg-loader/src/source/warg.rs

```diff
@@ -88,22 +94,26 @@ impl PackageSource for WargSource {
 
     async fn stream_content_unvalidated(
         &mut self,
-        _package: &PackageRef,
+        package: &PackageRef,
+        release: &Release,
+    ) -> Result<BoxStream<Result<Bytes, Error>>, Error> {
+        self.stream_content(package, release).await
+    }
+
+    async fn stream_content(
+        &mut self,
+        package: &PackageRef,
         release: &Release,
     ) -> Result<BoxStream<Result<Bytes, Error>>, Error> {
-        let digest = release
-            .content_digest
-            .to_string()
-            .parse()
-            .map_err(|err| ClientError::Other(anyhow::anyhow!("{err}")))?;
-        let stream = self
-            .api_client
-            .download_content(&digest)
-            .await
-            .map_err(ClientError::Api)?;
-        Ok(stream
-            .map_err(|err| Error::WargError(ClientError::Other(err)))
-            .boxed())
+        let package_name = package.try_into()?;
+
+        // warg client already validated the digest
+        let download = self
+            .client
+            .download_exact(&package_name, &release.version)
+            .await?;
+        let file = tokio::fs::File::open(download.path).await?;
+        Ok(ReaderStream::new(file).map_err(Into::into).boxed())
```

Comment:
Is this change necessary? It isn't always desirable to write to disk...

Reply:
I'll refactor to ignore the Warg cache

Reply:
Is it a problem that Warg caches the package logs? Or do you need entirely in-memory?

Reply:
The solution feels like adding an `InMemoryClient` instead of using `FileSystemClient` in `warg_client`. Thoughts?

Reply:
There is no (durable) writable disk available in this scenario; caching is handled at a higher level. You're probably right that a separate `InMemoryClient` would make sense as well, though its a little less important for the relatively small registry metadata.

Reply:
Should this crate CLI never use file system Warg download cache? Or should it be configuration?

Reply:
It just needs some path that doesn't require the entire package content to be written to disk.

Reply:
Fixed with the updated Warg 0.6.0